src/libPMacc/include/particles/traits/ResolveAliasFromSpecies.hpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright 2016 Heiko Burau
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "simulation_defines.hpp"
+#include "pmacc_types.hpp"
+#include "particles/memory/frames/Frame.hpp"
+#include "traits/GetFlagType.hpp"
+#include "traits/Resolve.hpp"
+#include "identifier/alias.hpp"
+
+namespace PMacc
+{
+namespace particles
+{
+namespace traits
+{
+
+/** Resolves a custom alias in the flag list of a particle species.
```

Comment:
It would be nice if you could provide a more detailed description of this trait. What kind of information it provides and how and where it is used.